internal/statemachine/classic.go

```diff
@@ -174,110 +217,123 @@ func (stateMachine *StateMachine) parseImageDefinition() error {
+					52,
+					errDetail,
+				),
+				errDetail,
+			)
 		}
-		// do custom validation for manual customization paths
-		if imageDefinition.Customization.Manual != nil {
-			jsonContext := gojsonschema.NewJsonContext("manual_path_validation", nil)
-			if imageDefinition.Customization.Manual.MakeDirs != nil {
-				for _, mkdir := range imageDefinition.Customization.Manual.MakeDirs {
-					// XXX: filepath.IsAbs() does returns true for paths like /../../something
-					// and those are NOT absolute paths.
-					if !filepath.IsAbs(mkdir.Path) || strings.Contains(mkdir.Path, "/../") {
-						errDetail := gojsonschema.ErrorDetails{
-							"key":   "customization:manual:mkdir:destination",
-							"value": mkdir.Path,
-						}
-						result.AddError(
-							imagedefinition.NewPathNotAbsoluteError(
-								gojsonschema.NewJsonContext("nonAbsoluteManualPath",
-									jsonContext),
-								52,
-								errDetail,
-							),
-							errDetail,
-						)
-					}
-				}
-			}
-			if imageDefinition.Customization.Manual.CopyFile != nil {
-				for _, copy := range imageDefinition.Customization.Manual.CopyFile {
-					// XXX: filepath.IsAbs() does returns true for paths like /../../something
-					// and those are NOT absolute paths.
-					if !filepath.IsAbs(copy.Dest) || strings.Contains(copy.Dest, "/../") {
-						errDetail := gojsonschema.ErrorDetails{
-							"key":   "customization:manual:copy-file:destination",
-							"value": copy.Dest,
-						}
-						result.AddError(
-							imagedefinition.NewPathNotAbsoluteError(
-								gojsonschema.NewJsonContext("nonAbsoluteManualPath",
-									jsonContext),
-								52,
-								errDetail,
-							),
-							errDetail,
-						)
-					}
-				}
-			}
-			if imageDefinition.Customization.Manual.TouchFile != nil {
-				for _, touch := range imageDefinition.Customization.Manual.TouchFile {
-					// XXX: filepath.IsAbs() does returns true for paths like /../../something
-					// and those are NOT absolute paths.
-					if !filepath.IsAbs(touch.TouchPath) || strings.Contains(touch.TouchPath, "/../") {
-						errDetail := gojsonschema.ErrorDetails{
-							"key":   "customization:manual:touch-file:path",
-							"value": touch.TouchPath,
-						}
-						result.AddError(
-							imagedefinition.NewPathNotAbsoluteError(
-								gojsonschema.NewJsonContext("nonAbsoluteManualPath",
-									jsonContext),
-								52,
-								errDetail,
-							),
-							errDetail,
-						)
-					}
-				}
+	}
+}
+
+// validateManualMakeDirs validates the Customization.Manual.MakeDirs section of the image definition
+func validateManualMakeDirs(imageDefinition *imagedefinition.ImageDefinition, result *gojsonschema.Result, jsonContext *gojsonschema.JsonContext) {
+	if imageDefinition.Customization.Manual.MakeDirs == nil {
+		return
+	}
+	for _, mkdir := range imageDefinition.Customization.Manual.MakeDirs {
+		// XXX: filepath.IsAbs() does returns true for paths like /../../something
+		// and those are NOT absolute paths.
```

Comment:
Correct. I have to admit I did not question this code when refactoring. The `filepath.IsAbs(path)` function is also simply checking if the path starts with `/`. Either the `filepath.IsAbs` implementation changed over time and we did not notice, or the other check (`strings.Contains(path, "/../")`) covers a use case I fail to see. Anyway, to make sure we can remove this I think I need to add test cases (because this is not really well covered) and I do not want to mix this with this PR.